stan/math/rev/core/var.hpp

```diff
@@ -77,14 +88,13 @@ class var_value<T, require_floating_point_t<T>> {
    * @param x Value of the variable.
    */
   template <typename S, require_convertible_t<S&, value_type>* = nullptr>
-  var_value(const S& x) : vi_(new vari_type(x, false)) {}  // NOLINT
+  var_value(S&& x) : vi_(new vari_type(std::forward<S>(x), false)) {}  // NOLINT
```

Comment:
We don't need perfect forwarding here if vari is making a copy anyway. When I suggested perfect forwarding, I assumed vari will be holding `Eigen::Matrix` so moving would be possible.

Reply:
Yeah my b. If we used `chainable_alloc` we could we could have Eigen matrices here. Though idk of the cost/benefit of that my priors feel like it would be slower

Reply:
Yes agree. Should we also have two separate constructors here, one for eigen and one for everything else, that wraps `x` into a `ref_type_t` so that if an expression comes in here it's only evaluated once?

Reply:
No, one constructor is fine. Wraping in `ref_type_t` would just make a redundant copy.

>> my priors feel like it would be slower

I think as soon as a model can use move instead of copy here on one big matrix it will outweight several destructor calls.